packages/ui/src/ui/pages/odin/controls/OdinOverview.tsx

```diff
@@ -36,29 +46,57 @@ import Link from '../../../components/Link/Link';
 import {YTDFDialog} from '../../../components/Dialog/Dialog';
 
 import ErrorBoundary from '../../../components/ErrorBoundary/ErrorBoundary';
-import {Dialog, Popup} from '@gravity-ui/uikit';
+import {Button, Dialog, Popup} from '@gravity-ui/uikit';
 import {setMetric} from '../_actions';
 import Icon from '../../../components/Icon/Icon';
 import OdinOverviewCreatePresetDialog from './OdinOverviewCreatePresetDialog';
 
 import {ODIN_CELL_SIZE, OdinTab} from '../odin-constants';
 
 import './OdinOverview.scss';
-import {absolute} from '../../../common/utils/url-ts';
+import {formatInterval} from '../../../components/common/Timeline/util';
+import WithStickyToolbar from '../../../components/WithStickyToolbar/WithStickyToolbar';
+import {useSearchParams} from '../../../hooks/use-search-params';
 
 const block = cn('odin-overview');
 
+const HALF_HOUR = 30 * 60 * 1000;
+
+let pollingIntervalId: NodeJS.Timeout | null | number = null;
+
+function startFetchDataPolling(fetchCb: () => void) {
+    if (pollingIntervalId) {
+        return;
+    }
+    pollingIntervalId = setInterval(fetchCb, 60 * 1000);
+}
+
+function stopFetchDataPolling() {
+    if (pollingIntervalId) {
+        clearInterval(pollingIntervalId);
+        pollingIntervalId = null;
+    }
+}
+
 function useOdinOverviewDataLoader(cluster: string) {
     const dispatch = useDispatch();
+    const searchParams = useSearchParams();
+
     React.useEffect(() => {
```

Comment:
The body of the effect should not be changed

Reply:
deleted